service/src/main/java/greencity/service/notification/NotificationServiceImpl.java

```diff
@@ -87,20 +91,43 @@ public void notifyUnpaidOrders() {
     }
 
     private boolean checkIfUnpaidOrderNeedsNewNotification(Order order, Optional<UserNotification> lastNotification) {
-        return (lastNotification.isEmpty()
-            || (lastNotification.get().getNotificationTime().isBefore(LocalDateTime.now(clock).minusDays(7))
-                || lastNotification.get().getNotificationTime().isEqual(LocalDateTime.now(clock).minusDays(7))))
-            && (order.getOrderDate().isAfter(LocalDateTime.now(clock).minusMonths(1))
-                || order.getOrderDate().isEqual(LocalDateTime.now(clock).minusMonths(1)))
-            && (order.getOrderDate().isBefore(LocalDateTime.now(clock).minusDays(3))
-                || order.getOrderDate().isEqual(LocalDateTime.now(clock).minusDays(3)));
+        if (lastNotification.isEmpty()) {
+            LocalDate now = LocalDate.now(clock);
+            LocalDateTime after3Days = order.getOrderDate();
+            LocalDate orderDate = after3Days.toLocalDate().plusDays(3);
+            return now.isEqual(orderDate);
+        }
+        LocalDate now = LocalDate.now(clock);
+        LocalDate nextNotification = lastNotification.get().getNextNotification().toLocalDate();
+        return now.isEqual(nextNotification);
```

Comment:
Should we also check if `now.isAfter(nextNotification)`
I see we send notifications once per day. 
We could have a bug when due for some reason(application crash, connection issue, some exeption) we did not send a notification and did not update the last modification time. The next day when we try to send a notification again and`lastNotification.get().getNextNotification().toLocalDate()` or `now.isEqual(orderDate.toLocalDate().plusDays(3))  will not be equal (it was equal yesterday) and we will not send notifications never

Reply:
You are right, but the requirement says that after 3 days an email will send, after that and email will send 3 times at the week (7 days), after 7 days, 3 emails more will send in 7 days.